File: artifacts/projects/test_s182_pycli2/source/todo_cli/commands.py
from todo_cli.storage import load_items, save_items


def _next_id(items):
    if not items:
        return 1
    return max(item["id"] for item in items) + 1


def cmd_add(title):
    title = title.strip()
    if not title:
        print("Error: title cannot be empty")
        return
    items = load_items()
    item = {
        "id": _next_id(items),
        "title": title,
        "completed": False,
    }
    items.append(item)
    save_items(items)
    print(f"Added: [{item['id']}] {item['title']}")


def cmd_list():
    items = load_items()
    if not items:
        print("No items.")
        return
    for item in items:
        status = "x" if item["completed"] else " "
        print(f"[{status}] {item['id']}: {item['title']}")


def cmd_complete(item_id):
    items = load_items()
    for item in items:
        if item["id"] == item_id:
            if item["completed"]:
                print(f"Item {item_id} is already complete.")
                return
            item["completed"] = True
            save_items(items)
            print(f"Completed: [{item_id}] {item['title']}")
            return
    print(f"Error: item {item_id} not found")


def cmd_delete(item_id):
    items = load_items()
    remaining = [i for i in items if i["id"] != item_id]
    if len(remaining) == len(items):
        print(f"Error: item {item_id} not found")
        return
    save_items(remaining)
    print(f"Deleted item {item_id}")

File: artifacts/projects/test_s182_gocli/source/cmd/commands.go
package cmd

import (
	"fmt"
	"os"

	"github.com/forge-demo/todo_gocli/storage"
)

func Add(store *storage.Store, title string) {
	item := storage.Item{
		ID:        store.NextID,
		Title:     title,
		Completed: false,
	}
	store.Items = append(store.Items, item)
	store.NextID++
	fmt.Printf("added task #%d: %s\n", item.ID, item.Title)
}

func List(store *storage.Store) {
	if len(store.Items) == 0 {
		fmt.Println("no tasks")
		return
	}
	for _, item := range store.Items {
		status := " "
		if item.Completed {
			status = "x"
		}
		fmt.Printf("[%s] #%d %s\n", status, item.ID, item.Title)
	}
}

func Complete(store *storage.Store, id int) {
	for i := range store.Items {
		if store.Items[i].ID == id {
			store.Items[i].Completed = true
			fmt.Printf("marked #%d complete\n", id)
			return
		}
	}
	fmt.Fprintf(os.Stderr, "task #%d not found\n", id)
	os.Exit(1)
}

func Delete(store *storage.Store, id int) {
	for i, item := range store.Items {
		if item.ID == id {
			store.Items = append(store.Items[:i], store.Items[i+1:]...)
			fmt.Printf("deleted task #%d\n", id)
			return
		}
	}
	fmt.Fprintf(os.Stderr, "task #%d not found\n", id)
	os.Exit(1)
}
